wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerPersonalAccessTokenFetcher.java

```diff
@@ -134,7 +134,8 @@ public Optional<Boolean> isValid(PersonalAccessToken personalAccessToken)
     }
     try {
       BitbucketPersonalAccessToken bitbucketPersonalAccessToken =
-          bitbucketServerApiClient.getPersonalAccessToken(personalAccessToken.getScmTokenId());
+          bitbucketServerApiClient.getPersonalAccessToken(
+              personalAccessToken.getScmTokenId(), personalAccessToken.getToken());
```

Comment:
It is hard to understand, why to retrieve personal access token we have to pass personal access token.

Reply:
Right, the `PersonalAccessToken` interface has a bit misleading name. In this case we implement an oauth token, but not a PAT.

Reply:
The Bitbucket Server oauth works in a bit different way: it uses the oauth token to create a PAT and then uses the PAT for all other operations.

Reply:
@tolusha Do you think we need to rename the `PersonalAccessToken` interface to `AccessToken` in order to avoid such confusion?

Reply:
Added a javadoc with a TODO.